soci/soci_index.go

```diff
@@ -330,32 +387,23 @@ func WriteSociIndex(ctx context.Context, indexWithMetadata IndexWithMetadata, st
 
 	log.G(ctx).WithField("digest", dgst.String()).Debugf("soci index has been written")
 
+	var refers *ocispec.Descriptor
+	if indexWithMetadata.Index.MediaType == OCIRefTypeManifestMediaType {
+		refers = indexWithMetadata.Index.Refers
+	} else {
+		refers = indexWithMetadata.Index.Subject
+	}
```

Comment:
How about also checking invalid combinations here (e.g. MediaaType is ORT but having Subject)?

Reply:
Hmm, I don't think it's necessary. This function is called after creating the index, so as long as the index creation logic is correct (which we verify through unit tests), this shouldn't be a problem. I've added a check for `refers == nil` though, which should return an errors on invalid combinations.